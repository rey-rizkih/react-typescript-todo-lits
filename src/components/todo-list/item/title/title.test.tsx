import { shallow } from "enzyme";
import { TodoTitle } from ".";

let wrapper;

const titleTestId = '[data-testid="todo-title"]';

describe("<TodoTitle/> rendering", () => {
  beforeEach(() => {
    wrapper = shallow(
      <TodoTitle data-testid="todo-title">Test Title</TodoTitle>
    );
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render one", () => {
    expect(wrapper.find(titleTestId)).toHaveLength(1);
  });

  it("shoudl render children correctly", () => {
    expect(wrapper.find(titleTestId).text()).toEqual("Test Title");
  });

  it("should have props isDone when isDone is true", () => {
    wrapper.setProps({ isDone: true });

    expect(wrapper.find(titleTestId).prop("isDone")).toBe(true);
  });
});
