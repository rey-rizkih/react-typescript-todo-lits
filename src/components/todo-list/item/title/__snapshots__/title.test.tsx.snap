// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoTitle/> rendering renders correctly 1`] = `
<Fragment>
  <Insertion
    cache={
      Object {
        "insert": [Function],
        "inserted": Object {},
        "key": "css",
        "nonce": undefined,
        "registered": Object {},
        "sheet": StyleSheet {
          "_insertTag": [Function],
          "before": null,
          "container": <head />,
          "ctr": 0,
          "insertionPoint": undefined,
          "isSpeedy": false,
          "key": "css",
          "nonce": undefined,
          "prepend": undefined,
          "tags": Array [],
        },
      }
    }
    isStringTag={false}
    serialized={
      Object {
        "map": undefined,
        "name": "16zvr0n",
        "next": undefined,
        "styles": "flex:1;padding:5px;border:none;font-size:20px;",
        "toString": [Function],
      }
    }
  />
  <ForwardRef(Typography)
    className="css-16zvr0n"
    data-testid="todo-title"
  >
    Test Title
  </ForwardRef(Typography)>
</Fragment>
`;

exports[`<TodoTitle/> renders correctly 1`] = `
<Fragment>
  <Insertion
    cache={
      Object {
        "insert": [Function],
        "inserted": Object {},
        "key": "css",
        "nonce": undefined,
        "registered": Object {},
        "sheet": StyleSheet {
          "_insertTag": [Function],
          "before": null,
          "container": <head />,
          "ctr": 0,
          "insertionPoint": undefined,
          "isSpeedy": false,
          "key": "css",
          "nonce": undefined,
          "prepend": undefined,
          "tags": Array [],
        },
      }
    }
    isStringTag={false}
    serialized={
      Object {
        "map": undefined,
        "name": "16zvr0n",
        "next": undefined,
        "styles": "flex:1;padding:5px;border:none;font-size:20px;",
        "toString": [Function],
      }
    }
  />
  <ForwardRef(Typography)
    className="css-16zvr0n"
    data-testid="todo-title"
  >
    Test Title
  </ForwardRef(Typography)>
</Fragment>
`;
